src/lib/storage/table.cpp

```diff
@@ -100,17 +102,54 @@ bool Table::column_nullable(const ColumnID column_id) const {
 
 std::shared_ptr<Chunk> Table::get_chunk(ChunkID chunk_id) {
   Assert(chunk_id < chunk_count(), "Chunk with ID does not exist.");
+  // We cannot hand out this chunk if compression is currently in progress, as somebody might modify it,
+  // causing us to lose this inserted data once the compression (which was based on the old chunk state) has finished.
+  // If the caller can ensure he won't modify the chunk, he can use the const overload instead.
+  const std::lock_guard<std::recursive_mutex> lock(_chunk_access_lock);
   return _chunks[chunk_id];
```

Comment:
This blocks access to all chunks, not just currently compressing chunks.
Also, this prevents compressing multiple chunks at the same time.

Not your fault, but:
Users of this function may choose to retain a copy of this pointer and later access it while it is being compressed.
This could lead to some segments containing rows, other segments do not. Again, not your fault but worth mentioning.

Reply:
Yeah, so apparently, this was a futile attempt to address the requirement of the task document.
Decided to remove this lock altogether and instead add a comment on why replacing the chunks should work fine without data races even without the lock.